File: study/src/study/day30.java
package study;

public class day30 {
    public String[] solution(String[] strings, int n) {
        for (int i = 0; i < strings.length-1; i++){
            for(int j = i; j < strings.length; j++ ){
                if (strings[i].substring(n, n+1).compareTo(strings[j].substring(n, n+1))>0){
                    String tmpStr = strings[i];
                    strings[i] = strings[j];
                    strings[j] = tmpStr;
                } else if (strings[i].substring(n, n+1).compareTo(strings[j].substring(n, n+1))==0){
                    if(strings[i].compareTo(strings[j])>0) {
                        String tmpStr = strings[i];
                        strings[i] = strings[j];
                        strings[j] = tmpStr;
                    }
                }

            }
        }
        return strings;
    }
    public static void main(String[] args){
        day30 a = new day30();
        String[] test = {"abce", "abcd", "cdx"};
        System.out.println(a.solution(test,2));
    }
}

File: study/bin/study/day30.py
def solution(strings, n):
    for i in range(len(strings)-1):
        for j in range(i, len(strings)):
            if (strings[i][n]>strings[j][n]):
                tmpStr = strings[i]
                strings[i] = strings[j]
                strings[j] = tmpStr
            elif (strings[i][n] == strings[j][n]):
                if (strings[i]>strings[j]):
                    tmpStr = strings[i]
                    strings[i] = strings[j]
                    strings[j] = tmpStr
    
    return strings

print(solution(["sun", "bed", "car"],1))

print("a">"b")